src/confirmation-window/ConfirmationWindow.tsx

```diff
@@ -26,11 +30,92 @@ const theme = createTheme({
 const ConfirmationWindow = () => {
     const { search } = useLocation();
     const query = useMemo(() => new URLSearchParams(search), [search]);
-    const data = useMemo(
+    const data: Record<string, string> = useMemo(
         () => JSON.parse(decodeURIComponent(query.get("params"))),
         [query],
     );
 
+    const [params, setParams] = useState<Record<string, string>>({});
+    const [loading, setLoading] = useState(false);
+
+    const processParams = async () => {
```

Comment:
This function should either be put directly in the below `useEffect` or wrapped in `useCallback`.

Reply:
Also I think I asked you to move state management from this component into a dedicated context.

Reply:
This is not supposed to be used anywhere else, so there's of no use to put it into the context. So I've moved it to the hook for better code splitting.